spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultPublisherFactory.java

```diff
@@ -70,8 +69,7 @@ public DefaultPublisherFactory(GcpProjectIdProvider projectIdProvider,
 		Assert.notNull(projectIdProvider.getProjectId(),
 				"The project ID provider can't return null.");
 
-		this.projectId = projectIdProvider.getProjectId()
-				.orElseThrow(MissingProjectIdException::new);
+		this.projectId = projectIdProvider.getProjectId();
```

Comment:
Now let's `Assert.hasText(this.projectId)` as well to nail your initial idea with just removed exception!